### PR TITLE
Introduce SlackChannelIntegrations + Add attribution to Slack search results

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster @eishay

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/util/DescriptionElements.scala [hunk 1]
```diff
@@ -62,6 +64,9 @@ object DescriptionElements {
   implicit def fromDollarAmount(v: DollarAmount): BasicElement = v.toDollarString
   implicit def fromRole(role: OrganizationRole): BasicElement = role.value
 
+  private val prettyTime = new PrettyTime()
+  implicit def fromDateTime(time: DateTime): BasicElement = prettyTime.format(time.toDate)
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/common/util/DescriptionElements.scala:68]: @ryanpbrewster

File: kifi-backend/modules/common/app/com/keepit/model/KeepSourceAttribution.scala [hunk 2]
```diff
@@ -18,6 +18,7 @@ import scala.util.{ Failure, Success, Try }
   // https://groups.google.com/forum/#!topic/twitter-development-talk/ahbvo3VTIYI
   override def toString = id.toString
 }
+//todo: flip references i.e. kill Keep.attributionId and introduce KeepSourceAttribution.keepId
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/model/KeepSourceAttribution.scala:21]: @andrewconner @ryanpbrewster would remove a field from `Keep`, and usually we're dealing in `Id[Keep]`, not want to expose `Id[KeepSourceAttribution]`. I'm assuming this was just an old bad decision. What do you think?

Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/KeepSourceAttribution.scala:21]: :+1: seems good to me

File: kifi-backend/modules/common/app/com/keepit/common/util/DescriptionElements.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/common/util/DescriptionElements.scala:68]: :ok_hand:

**Pull request merged**